engine/crates/engine-v2/src/response/read/ser.rs

```diff
@@ -105,9 +108,32 @@ impl<'a> serde::Serialize for SerializableError<'a> {
         if let Some(ref path) = self.error.path {
             map.serialize_entry("path", &SerializableResponsePath { keys: self.keys, path })?;
         }
-        if !self.error.extensions.is_empty() {
-            map.serialize_entry("extensions", &self.error.extensions)?;
+        map.serialize_entry(
+            "extensions",
+            &SerializableExtension {
+                code: self.error.code,
+                extensions: &self.error.extensions,
+            },
+        )?;
+        map.end()
+    }
+}
+
+struct SerializableExtension<'a> {
+    code: ErrorCode,
+    extensions: &'a [(Cow<'static, str>, serde_json::Value)],
+}
+
+impl<'a> serde::Serialize for SerializableExtension<'a> {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: serde::Serializer,
+    {
+        let mut map = serializer.serialize_map(Some(self.extensions.len() + 1))?;
+        for (key, value) in self.extensions {
+            map.serialize_entry(key, value)?;
         }
+        map.serialize_entry("code", &self.code)?;
```

Comment:
What happens if a hook adds a code to the error extensions. Do we get two codes in the response?

Reply:
gets ignored today, I should probably just let it through though